codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/customizations/serde/SerializeImplGenerator.kt

```diff
@@ -57,10 +56,11 @@ import software.amazon.smithy.rust.codegen.core.util.isEventStream
 import software.amazon.smithy.rust.codegen.core.util.isTargetUnit
 import software.amazon.smithy.rust.codegen.core.util.letIf
 import software.amazon.smithy.rust.codegen.core.util.toPascalCase
-import software.amazon.smithy.rust.codegen.server.smithy.ServerRustSettings
-import software.amazon.smithy.rust.codegen.server.smithy.hasConstraintTrait
 
-class SerializeImplGenerator(private val codegenContext: CodegenContext) {
+class SerializeImplGenerator(
+    private val codegenContext: CodegenContext,
+    private val unwrapConstraints: (Shape) -> Writable,
```

Comment:
this this is now a public name, we should probably name it better. It's not actually "unwrapping" constraints in that it doesn't fail if the constraints are violated. Maybe `constrainedShapeToInner` or something?